From c529aa85eb305266ee2166460b1fa8216ccbc8e5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Mislav=20Marohni=C4=87?= <mislav.marohnic@gmail.com>
Date: Mon, 14 Sep 2009 15:44:42 +0200
Subject: [PATCH] change `starts/ends_with?` implementations for ruby 1.8.7 compatibility

`starts/ends_with?` methods shouldn't cast argument to string because ruby 1.8.7
doesn't seem to do that. for example:

    "foobar".ends_with?(:bar)
    # => true in ActiveSupport implementation, false in ruby 1.8.7
---
 .../core_ext/string/starts_ends_with.rb            |    2 --
 activesupport/test/core_ext/string_ext_test.rb     |    2 ++
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/activesupport/lib/active_support/core_ext/string/starts_ends_with.rb b/activesupport/lib/active_support/core_ext/string/starts_ends_with.rb
index 09f9a18..ec23282 100644
--- a/activesupport/lib/active_support/core_ext/string/starts_ends_with.rb
+++ b/activesupport/lib/active_support/core_ext/string/starts_ends_with.rb
@@ -20,13 +20,11 @@ module ActiveSupport #:nodoc:
 
         # Does the string start with the specified +prefix+?
         def starts_with?(prefix)
-          prefix = prefix.to_s
           self[0, prefix.length] == prefix
         end
 
         # Does the string end with the specified +suffix+?
         def ends_with?(suffix)
-          suffix = suffix.to_s
           self[-suffix.length, suffix.length] == suffix      
         end
       end
diff --git a/activesupport/test/core_ext/string_ext_test.rb b/activesupport/test/core_ext/string_ext_test.rb
index e57008b..addbc39 100644
--- a/activesupport/test/core_ext/string_ext_test.rb
+++ b/activesupport/test/core_ext/string_ext_test.rb
@@ -177,6 +177,7 @@ class StringInflectionsTest < Test::Unit::TestCase
     s = "hello"
     assert s.starts_with?('h')
     assert s.starts_with?('hel')
+    assert !s.starts_with?(:hel)
     assert !s.starts_with?('el')
 
     assert s.start_with?('h')
@@ -185,6 +186,7 @@ class StringInflectionsTest < Test::Unit::TestCase
 
     assert s.ends_with?('o')
     assert s.ends_with?('lo')
+    assert !s.ends_with?(:lo)
     assert !s.ends_with?('el')
 
     assert s.end_with?('o')
-- 
1.6.4.1

